pkg/kritis/admission/admission.go

```diff
@@ -57,88 +62,190 @@ var (
 	defaultViolationStrategy = violation.LoggingStrategy{}
 )
 
-// This admission controller looks for the breakglass annotation
-// If one is not found, it validates against image security policies
-// TODO: Check for attestations
+var (
+	runtimeScheme = runtime.NewScheme()
+	codecs        = serializer.NewCodecFactory(runtimeScheme)
+)
+
 func AdmissionReviewHandler(w http.ResponseWriter, r *http.Request) {
-	glog.Info("Starting admission review handler version %s ...", version.Commit)
-	pod, err := admissionConfig.retrievePod(r)
+	glog.Infof("Starting admission review handler, version: %s",
+		version.Commit)
+	body, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		glog.Error(err)
-		w.WriteHeader(http.StatusBadRequest)
+		glog.Errorf("Error reading body: %v", err)
+		http.Error(w, "can't read body", http.StatusBadRequest)
 		return
 	}
-	// First, check for a breakglass annotation on the pod
-	if checkBreakglass(pod) {
-		glog.Infof("found breakglass annotation, returning successful status")
-		returnStatus(constants.SuccessStatus, constants.SuccessMessage, w)
-		return
+
+	ar := v1beta1.AdmissionReview{}
+	deserializer := codecs.UniversalDeserializer()
+	if _, _, err := deserializer.Decode(body, nil, &ar); err != nil {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusUnauthorized)
+
+		payload, err := json.Marshal(&v1beta1.AdmissionResponse{
+			UID:     ar.Request.UID,
+			Allowed: false,
+			Result: &metav1.Status{
+				Status:  string(constants.FailureStatus),
+				Message: err.Error(),
+			},
+		})
+		if err != nil {
+			glog.Info(err)
+		}
+		w.Write(payload)
+	}
+
+	admitResponse := &v1beta1.AdmissionReview{
+		Response: &v1beta1.AdmissionResponse{
+			UID:     ar.Request.UID,
+			Allowed: true,
+			Result: &metav1.Status{
+				Status:  string(constants.SuccessStatus),
+				Message: constants.SuccessMessage,
+			},
+		},
+	}
+
+	if ar.Request.Kind.Kind == "Deployment" {
```

Comment:
we're going to want to expand this, maybe refactor it to a map of string to reviewer functions?

Reply:
done